nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -88,11 +89,24 @@ public void initialize(String name, Map<String, String> options) {
         options.get(removePrefix.apply(NessieConfigConstants.CONF_NESSIE_REF));
     String requestedHash =
         options.get(removePrefix.apply(NessieConfigConstants.CONF_NESSIE_REF_HASH));
-    NessieApiV1 api =
+
+    NessieClientBuilder<?> nessieClientBuilder =
         createNessieClientBuilder(
                 options.get(NessieConfigConstants.CONF_NESSIE_CLIENT_BUILDER_IMPL))
-            .fromConfig(x -> options.get(removePrefix.apply(x)))
-            .build(NessieApiV1.class);
+            .fromConfig(x -> options.get(removePrefix.apply(x)));
+    final String apiVersion = options.get(NessieUtil.CLIENT_API_VERSION);
+    NessieApiV1 api;
+    if (apiVersion == null || apiVersion.equalsIgnoreCase("v1")) {
```

Comment:
I wonder if a simple `int` value is preferable. Nessie Client API version checks use simple ints, e.g. https://github.com/projectnessie/nessie/pull/6818

Reply:
ok changed to just 1 or 2. 

Even Iceberg format version is 1 or 2. So, the syntax is inline.